### PR TITLE
materialized: fix + tweak startup logging

### PR DESCRIPTION
**Pull request by benesch**
Details in individual commit messages. Lots of code movement, so much easier to review commit by commit.

---
This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/4647)

### CODE REVIEW

File: src/materialized/src/bin/materialized/sys.rs [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+        }
+    };
+    trace!("initial nofile rlimit: ({}, {})", soft, hard);
+
+    #[cfg(target_os = "macos")]
+    let hard = {
+        use ore::cast::CastFrom;
+        use sysctl::Sysctl;
+
+        // On macOS, getrlimit by default reports that the hard limit is
```
Comment (ruchirK) [inline src/materialized/src/bin/materialized/sys.rs:43]: Is there a source for this comment? I don't doubt it at all but this is the sort of thing that if I was reading a year from now I'd wonder how the author knew that

File: src/materialized/src/bin/materialized/main.rs [hunk 2]
```diff
@@ -379,7 +381,7 @@ environment:{}",
         env_message
     );
 
-    adjust_rlimits();
+    sys::adjust_rlimits();
```
Comment (ruchirK) [inline src/materialized/src/bin/materialized/main.rs:414]: Is this commit mainly to organize this crate better and not have it balloon into one giant file? If not, then I'm not sure I understand the purpose for this commit

**Review by ruchirK — APPROVED** [review]
This all looks great to me!

File: src/materialized/src/bin/materialized/sys.rs (hunk 1 above)
Comment (benesch) [reply src/materialized/src/bin/materialized/sys.rs:44]: https://github.com/cockroachdb/cockroach/commit/e777ba081ac1fd9930420aecdb1638e816894cb0 :D

File: src/materialized/src/bin/materialized/main.rs (hunk 2 above)
Comment (benesch) [reply src/materialized/src/bin/materialized/main.rs:414]: Yep, exactly!

**Comment by benesch** [comment]
Thanks for the review!

**Pull request merged**